src/Files.App/Helpers/NavigationHelpers.cs

```diff
@@ -330,6 +331,27 @@ private static async Task<FilesystemResult> OpenDirectory(string path, IShellPag
 
 				opened = (FilesystemResult)true;
 			}
+			else if (isNetwork)
+			{
+				var auth = await NetworkDrivesAPI.AuthenticateNetworkShare(path);
```

Comment:
Perhaps we can move the call to AuthenticateNetworkShare() to ItemViewModel.cs. I'm seeing that showing the credentials dialog for FTP connections is handled there. This way we could remove both this logic and the one added to ToolbarViewModel.cs.

Reply:
Excellent thought, @radu-cernatescu do you want to make this change in a separate PR?

Reply:
@gave92 & @yaira2,
For sure, I'll make these changes and create a new PR for this.

Reply:
PR #11193  is up! I guess this conversation can be resolved. Thanks